Embrace Unity iOS Interface/EmbraceManager.swift

```diff
@@ -0,0 +1,488 @@
+        } else {
+            span.addEvent(name: name,
+                          attributes: attributes,
+                          timestamp: convertDoubleToDate(ms: time))
+        }
+        
+        return true
+    }
+    
+    static func addSpanAttributeToSpan(spanId: String, key: String, value: String) -> Bool {
+        guard let span = spanRepository.get(spanId: spanId) else {
+            return false
+        }
+        
+        span.setAttribute(key: key, value: value)
+        Embrace.client?.flush(span)
+        
+        return true
+    }
+    
+    static func recordCompletedSpan(
+        name: String,
+        startTimeMs: Double,
+        endTimeMs: Double,
+        errorCodeString: String,
+        parentSpanId: String,
+        attributes: inout [String: String],
+        events: [RecordingSpanEvent]) -> Bool {
+            
+            var parent = parentSpanId.isEmpty ? spanRepository.get(spanId: parentSpanId) : nil
+            attributes.updateValue("true", forKey: "emb.key")
+            
+            if Embrace.client == nil {
+                return false
+            }
+            
+            Embrace.client?.recordCompletedSpan(name: name,
+                                                type: .performance,
+                                                parent: parent,
+                                                startTime: convertDoubleToDate(ms: startTimeMs),
+                                                endTime: convertDoubleToDate(ms: endTimeMs),
+                                                attributes: attributes,
+                                                events: events,
+                                                errorCode: convertStringToErrorCode(str: errorCodeString))
+            return true
+    }
+    
+    // TODO: Reduce code duplication between handled and unhandled exceptions
+    static func logHandledException(name: String, message: String, stacktrace: String) {
+        var attributes = [
+            "exception.stacktrace": stacktrace,
+            "emb.exception_handling": "handled",
+            "emb.type": "sys.exception",
+            "exception.message": message,
+            "exception.type": name
+        ]
+        
+        Embrace.client?.log("Unity exception",
+                            severity: .error,
+                            timestamp: Date(), // Should we let users input their own exception timestamp?
+                            attributes: attributes)
+    }
+    
+    static func logUnhandledException(name: String, message: String, stacktrace: String) {
+        var attributes = [
+            "exception.stacktrace": stacktrace,
+            "emb.exception_handling": "unhandled",
+            "emb.type": "sys.exception",
+            "exception.message": message,
+            "exception.type": name
+        ]
+        
+        Embrace.client?.log("Unity exception",
```

Comment:
not sure if the behaviour is different here but on RN we did have to include an extra flag to prevent the iOS SDK from automatically including a native ios stack trace in this call as well: https://github.com/embrace-io/embrace-react-native-sdk/blob/main/packages/core/ios/RNEmbrace/EmbraceManager.swift#L753

Before we set the flag we were seeing 2 separate stack traces in the log, one for JS and one for Swift

Reply:
That makes sense. Right.